arch/src/x86_64/mod.rs

```diff
@@ -202,6 +209,22 @@ fn add_e820_entry(
     Ok(())
 }
 
+/// Returns the memory address where the initrd could be loaded.
+pub fn initrd_load_addr(guest_mem: &GuestMemoryMmap, initrd_size: usize) -> super::Result<u64> {
+    let first_region = guest_mem
+        .find_region(GuestAddress::new(0))
+        .ok_or(super::Error::InitrdAddress)?;
+    // It's safe to cast to usize because the size of a region can't be greater than usize.
+    let lowmem_size = first_region.len() as usize;
+
+    if lowmem_size < initrd_size {
+        return Err(super::Error::InitrdAddress);
+    }
+
```

Comment:
Overall, this makes a couple of assumptions:
1. The first memory region will be under the 4G limit (lowmem)
1. There is enough space left at the end of that region to load the initramfs image

The first one is a safe assumption to make on x86, the second one is ok because we can safely assume the image will be at most a few dozen MBs.

Reply:
should I add this comment to the function?